CTRESim-java/src/main/java/com/ctre/phoenix/motorcontrol/can/WPI_VictorSPX.java

```diff
@@ -0,0 +1,205 @@
+import edu.wpi.first.wpilibj.smartdashboard.SendableRegistry;
+
+public class WPI_VictorSPX implements SpeedController, Sendable {
+  private String _description;
+  
+  private double _speed;
+  
+  private SimDevice m_simDevice;
+  
+  private SimDouble m_simSpeed;
+  
+  private SimBoolean m_simInvert;
+  
+  public static final double kDefaultSafetyExpiration = 0.1D;
+  
+  private WPI_MotorSafetyImplem _motorSafety = null;
+  
+  private final Object _lockMotorSaf = new Object();
+  
+  private double _motSafeExpiration = 0.1D;
+  
+  private String m_name;
+  
+  private String m_subsystem;
+  
+  public WPI_VictorSPX(int deviceNumber) {
+    this.m_name = "";
+    this.m_subsystem = "Ungrouped";
+    HAL.report(67, deviceNumber + 1);
+    this._description = "Victor SPX " + deviceNumber;
+    SendableRegistry.addLW(this, "Victor SPX ", deviceNumber);
+    this.m_simDevice = SimDevice.create("Victor SPX", deviceNumber);
+    if (this.m_simDevice != null) {
+      this.m_simSpeed = this.m_simDevice.createDouble("Motor Output", false, 0.0D);
+      this.m_simInvert = this.m_simDevice.createBoolean("Inverted?", false, false);
+    } 
+  }
+  
+  /**
+   * ALTERED FOR USAGE (Team 1418)
+   */
+  public void follow(Object masterToFollow, Object followerType) {
+    return;
+  }
+
+  /**
+   * ALTERED FOR USAGE (Team 1418)
+   */
+  public void follow(Object masterToFollow) {
+    follow(masterToFollow, null);
+  }
+
+  /**
+   * ALTERED FOR USAGE (Team 1418)
+   */
+  public void setInverted(InvertType invertType) {
+    // this._invert = invertType;
+  }
+
+  public void set(double speed) {
+    this._speed = speed;
+    set(ControlMode.PercentOutput, this._speed);
+    feed();
+  }
+
+  public void pidWrite(double output) {
+    set(output);
+  }
+  
+  public double get() {
+    if (this.m_simSpeed != null)
+      return this.m_simSpeed.get(); 
+    return this._speed;
+  }
+  
+  public void set(ControlMode mode, double value) {
+    feed();
+    simSet(mode, value);
+  }
+  
+  @Deprecated
+  public void set(ControlMode mode, double demand0, double demand1) {
+    feed();
+    simSet(mode, demand0);
+  }
+  
+  public void set(ControlMode mode, double demand0, DemandType demand1Type, double demand1) {
+    feed();
+    simSet(mode, demand0);
+  }
+  
+  private void simSet(ControlMode mode, double demand0) {
+    if (this.m_simSpeed != null && this.m_simInvert != null) {
+      switch (mode) {
+        case PercentOutput:
+          this.m_simSpeed.set(demand0 * (this.m_simInvert.get() ? -1 : 1));
+          return;
+      } 
+      this.m_simSpeed.set(0.0D);
+    } 
+  }
+  
+  public void setVoltage(double outputVolts) {
+    set(outputVolts / RobotController.getBatteryVoltage());
+  }
+  
+  public void setInverted(boolean isInverted) {
+    if (this.m_simInvert != null)
+      this.m_simInvert.set(isInverted); 
+  }
+  
+  public boolean getInverted() {
+    if (this.m_simInvert != null)
+      return this.m_simInvert.get(); 
```

Comment:
I tend to not like `m_` syntax... is it conventional for Java robot code?

Reply:
Just to be clear: not a problem at all if this is conventional.

Reply:
The like it in wpilib. I hate it everywhere. I let my ide color my variables to tell me what they are :joy:

Reply:
Very sad.

Reply:
Yeah it's quite annoying. I'm going to leave it in this case since it's from the CTRESim-java package, so it's directly from decompiled phoenix code.